### PR TITLE
Feature/login integration

### PR DESCRIPTION
**Pull request by davewish**

### CODE REVIEW

File: src/components/Drawer.jsx [hunk 1]
```diff
@@ -147,6 +147,8 @@ const ChurchDrawer = React.memo(function ChurchDrawer({ handleDrawerClose, drawe
       variant="persistent"
       anchor="left"
       open={open}
+      onClose={handleDrawerClose}
+      onOpen={() => {}}
```
Comment (kuldep7) [inline src/components/Drawer.jsx:151]: Why we are sending onOpen like this ?

File: src/constants/endpoints.json [hunk 2]
```diff
@@ -1,10 +1,12 @@
 {
   "test": {
     "globalSetting": "global_setting",
-    "contentData": "contents?content_category=Home&start=0&lang=english"
+    "contentData": "contents?content_category=Home&start=0&lang=english",
+    "login": "login"
```
Comment (kuldep7) [inline src/constants/endpoints.json:5]: do we need to hardcode the query params ?

File: src/hooks/useLogin.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+const useLogin = () => {
+  const { setAuthState } = useAuth();
+  const axiosPrivate = useAxiosPrivate();
+  const login = async ({ email, password }) => {
+    const formData = new FormData();
+    formData.append("email", email);
+    formData.append("password", password);
+    const response = await axiosPrivate.post(`/api/${currentConfig.login}`, formData, {
+      headers: {
+        "Content-Type": "multipart/form-data"
```
Comment (kuldep7) [inline src/hooks/useLogin.js:16]: on login why do we need multipart ?

**Review by kuldep7 — CHANGES_REQUESTED** [review]

File: src/components/Drawer.jsx (hunk 1 above)
Comment (davewish) [reply src/components/Drawer.jsx:151]: Actually ,it was not my task , onOpen is required props  Component, I added for  remove   console.log warning

File: src/constants/endpoints.json (hunk 2 above)
Comment (davewish) [reply src/constants/endpoints.json:5]: It should not be , but I did for time being

File: src/hooks/useLogin.js (hunk 3 above)
Comment (davewish) [reply src/hooks/useLogin.js:16]: > on login why do we need multipart ?
API 's request body   content type  
is multipart

Comment (davewish) [reply src/hooks/useLogin.js:16]: > and is it required to have a separate hook for login ?

My thinking process when I did  was  to separate  the logic part from the UI  even though code is not shared an other part of the code 2 ,  I will put in the custom hook  other login related use cases like registration

Comment (kuldep7) [reply src/hooks/useLogin.js:16]: Okay so we can use a more generic name for it then

**Review by misghna — APPROVED** [review]

**Review by misghna — APPROVED** [review]

**Pull request merged**